test/e2e_node/critical_pod_test.go

```diff
@@ -78,15 +76,24 @@ var _ = framework.KubeDescribe("CriticalPod [Serial] [Disruptive] [NodeFeature:C
 					v1.ResourceMemory: resource.MustParse("100Mi"),
 				},
 			})
+
 			nonCriticalBestEffort := getTestPod(false, bestEffortPodName, v1.ResourceRequirements{})
+			capacity := getNodeCPUAndMemoryCapacity(f)
+			used := v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("200m"),
+				v1.ResourceMemory: resource.MustParse("200Mi"),
+			}
+			available := getAvailableCPUAndMemoryCapacity(capacity, used)
 			criticalPod := getTestPod(true, criticalPodName, v1.ResourceRequirements{
 				// request the entire resource capacity of the node, so that
 				// admitting this pod requires the other pod to be preempted
-				Requests: getNodeCPUAndMemoryCapacity(f),
+				Requests: available,
 			})
-
-			_, err = f.ClientSet.SchedulingV1().PriorityClasses().Create(systemCriticalPriority)
-			gomega.Expect(err == nil || errors.IsAlreadyExists(err)).To(gomega.BeTrue(), "failed to create PriorityClasses with an error: %v", err)
+			// SystemPriorityClasses define system priority classes that are auto-created at cluster bootstrapping.
+			// SystemNodeCritical used for system critical pods that must not be moved from their current node.
+			// Because e2e_node.test does not start the scheduler, SystemNodeCritical needed to be created manually.
+			_, err = f.ClientSet.SchedulingV1().PriorityClasses().Create(schedulingv1.SystemPriorityClasses()[0])
```

Comment:
should this loop over all items in `f.ClientSet.SchedulingV1().PriorityClasses()` and create, tolerating `AlreadyExists` errors?

Reply:
Agree!

The scheduler create them here: https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/scheduling/rest/storage_scheduling.go#L125-L141, so I agree that test should do the same.